presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/NodeSelectionStats.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.scheduler.nodeSelection;
+
+import org.weakref.jmx.Managed;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+@ThreadSafe
+public class NodeSelectionStats
```

Comment:
This class indicates hard assumption on "we only have 2 preferred nodes and all other selections are random". I'm ok with this for now. @shixuan-fan, do you have better suggestions?

Reply:
I share the similar concern and it does not seem to fit the angle from engine's perspective, as 2 preferred nodes are something very hive-specific. 

If `weakref` could generate different stats for elements in a list, that's great, but I highly doubt that is possible.

The other way is we only take primary, non primary, and random (seems to be the easiest fix).

One final random thought is: since 2 preferred nodes are hive-specific, I'm curious if this is something that could be done in connector. For example, engine passed the decision back to connector, and let the connector collect the stats? But this is a bit weird because node selection is done in the engine.

I personally think the most feasible one is to change secondary to non-primary.

Reply:
Agree. Let's change secondary to non-primary